### PR TITLE
[RFR] Minimal test suite

### PR DESCRIPTION
**Pull request by Kmaschta**
Introduce E2E test which run the API and use it through the command.
The application is tested with real input and ouput sent to the client.

## Details
Since we only test client output, we don't need to load fixtures or reset it each time.
Instead, all the commands are run in a specific folder (`test/.env`) which is deleted between all the tests.

## TODO
- [x] More tests for the project init (test that the environments are created, etc)
- [x] Test available get formats (JSON, YML, env vars)
- [x] Test environment management (get, list, create)
- [x] Test configuration version list (`comfy ls`)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,12 +1,16 @@
+.PHONY: test
+
 install:
 	cd api && make install
```
Comment (jpetitcolas) [inline Makefile]: Use `make -C api install` instead.

File: test/Makefile [hunk 2]
```diff
@@ -0,0 +1,45 @@
+export CI ?= false
+export PM2_HOME ?= .pm2
+
+install:
+	npm install
+
+run-test-api:
+	if [ "${CI}" = "false" ]; then \
+		echo 'Starting test database...'; \
+		cd ../api && make install-db || true; \
```
Comment (jpetitcolas) [inline test/Makefile]: Use `;` which will execute next condition either if first one succeeded or failed.

File: test/Makefile [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+	rm -rf ./.env/*
+
+teardown-test: stop-test-api
+	./node_modules/.bin/pm2 kill
+
+test: setup-test
+	@echo 'Starting test suite...'
+	./node_modules/.bin/mocha --require co-mocha \
+		./setup.js \
+		'./specs/**/*.js'
```
Comment (jpetitcolas) [inline test/Makefile]: Use a `find` subshell here to prevent some issues with deeply nested tests.

File: test/setup.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        if (deleteFolder) {
+            yield cb => fs.rmdir(dir, cb);
+        }
+    }
+};
+
+
+afterEach(function* () {
+    // Delete all .env folder content
+    yield deleteFolderContent('./.env/');
```
Comment (jpetitcolas) [inline test/setup.js]: Can't we do it directly in the `Makefile` instead, using good old Bash?

Comment (Kmaschta) [inline test/setup.js]: It should be done after each test, the Makefile can't run a command between each mocha test

File: test/Makefile (hunk 3 above)
Comment (Kmaschta) [inline test/Makefile]: Nope, I explained you around a beer

File: Makefile (hunk 1 above)
Comment (Kmaschta) [inline Makefile]: Nice catch, I didn't know

**Comment by Kmaschta** [comment]
The tests are slow, and I just added a timeout to fix them.
But I created an issue for this: https://github.com/marmelab/comfygure/issues/91

File: Makefile [hunk 5]
```diff
@@ -1,12 +1,16 @@
+.PHONY: test
+
 install:
-	cd api && make install
-	cd cli && make install
-	cd admin && make install
+	make -C api install
```
Comment (ThieryMichel) [inline Makefile]: did'nt know this one :+1:

**Pull request merged**